src/flash/display/LoaderInfo.ts

```diff
@@ -72,24 +75,27 @@ module Shumway.AVM2.AS.flash.display {
       this._mappedSymbolsLoaded = 0;
       this._fontsLoaded = 0;
       this._avm1Context = null;
-
       this._colorRGBA = 0xFFFFFFFF;
     }
 
-    setFile(file: SWFFile) {
+    setFile(file: any /* SWFFile | ImageFile */) {
       release || assert(!this._file);
       this._file = file;
-      // TODO: remove these duplicated fields from LoaderInfo.
       this._bytesTotal = file.bytesTotal;
-      this._swfVersion = file.swfVersion;
-      this._frameRate = file.frameRate;
-      var bbox = file.bounds;
-      this._width = bbox.xMax - bbox.xMin;
-      this._height = bbox.yMax - bbox.yMin;
-      this._colorRGBA = file.backgroundColor;
-
-      if (!file.attributes || !file.attributes.doAbc) {
-        this._actionScriptVersion = ActionScriptVersion.ACTIONSCRIPT2;
+      if (file instanceof SWFFile) {
+        // TODO: remove these duplicated fields from LoaderInfo.
```

Comment:
That + throwing "The loading object is not sufficiently loaded to provide this information" errors in their getters if this._file is not set.